[spellcheck] Use behavior instead of behaviour

3nids opened the pull request:
Standing for using "behavior" instead of behaviour" implies some API changes and I am not really sure they're worth it.

The following would be renamed:
* in QgsVectorLayer, the enum `SelectBehaviour`
* in QgsColorButton, the enum `Behaviour` and its corresponding setter/getter
* in QgsComposerTableV2, the enum `WrapBehaviour`
* in processing, the argument of QGIS' SnapGeometries argument `BEHAVIOUR`

Thoughts?

3nids committed:
[spelling] replace behaviour by behavior

nyalldawson commented:
I think the API breaks should be made. Part of the intention with the 3.0 API changes is to make the API more predictable and consistent, and fixing these inconsistent spelling variations is a big part of this. So +1 to break here.

DelazJ commented on python/plugins/processing/algs/qgis/SnapGeometries.py:
```diff
@@ -41,7 +41,7 @@ class SnapGeometriesToLayer(GeoAlgorithm):
     TOLERANCE = 'TOLERANCE'
     OUTPUT = 'OUTPUT'
-    BEHAVIOUR = 'BEHAVIOUR'
+    BEHAVIOR = 'BEHAVIOUR'
```
did you miss this one?

m-kuhn commented:
I agree, +1 for breaks for a predictable api

3nids commented:
@DelazJ thanks, I fixed this

@nyalldawson @m-kuhn I updated the api_break.dox.

More changes that might need to be discussed:

* in analysis: Triangulation::forcedCrossBehavior (enum) and setForcedCrossBehavior (should be documented?)
* in QgsComposerArrow mBoundsBehavior attribute (project break)
* /Projections/defaultBehaviour setting (application behavior break)

nyalldawson commented:
> in analysis: Triangulation::forcedCrossBehavior (enum) and setForcedCrossBehavior (should be documented?)

Break this - the enum should be ForcedCrossBehavior (initial capital). I'd also rename all these enum values as they are a mash of capitalization and camelcase/underscore, and mispell vertex as vertice. setForcedCrossBehaviour should be setForcedCrossBehavior (and a getter should ideally be added too!).

> in QgsComposerArrow mBoundsBehavior attribute (project break)

I'd remove this pre 2.2 project compatibility code totally. 2.2 is ancient and we can always recommend that people open and save old projects in 2.18 before loading into 3.x if this is an issue. We've removed other old project compatibility from 3.0 (eg old labeling, symbology) and I think this also applies. I'm also about to remove compatibility with <= 1.8 data defined label settings.

> /Projections/defaultBehaviour setting (application behavior break)

No issue there - qgis 3.0 has a new settings folder to allow things like this.

3nids commented:
@nyalldawson thanks for the feedback. I update this accordingly.

I have some processing tests failing and I don't understand why. Could you give me a hint here?

nyalldawson commented on python/analysis/interpolation/DualEdgeTriangulation.sip:
```diff
@@ -40,8 +40,8 @@ class DualEdgeTriangulation: Triangulation
-    /** Sets the behaviour of the triangulation in case of crossing forced lines*/
-    virtual void setForcedCrossBehaviour( Triangulation::forcedCrossBehaviour b );
+    /** Sets the behavior of the triangulation in case of crossing forced lines*/
+    virtual void setForcedCrossBehavior( Triangulation::forcedCrossBehavior b );
```
I think here's your error (also TriDecorator.sip) - Triangulation::forcedCrossBehavior is missing the capital for Forced

3nids replied:
Thanks a lot!

3nids commented:
merged in 748be8d
